File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/migration/MigrationExchangeRequest.java
package com.github.seratch.jslack.api.methods.request.migration;

import com.github.seratch.jslack.api.methods.SlackApiRequest;

import java.util.List;

/**
 * For Enterprise Grid workspaces, map local user IDs to global user IDs
 */
public class MigrationExchangeRequest implements SlackApiRequest {

    /**
     * Authentication token. Requires scope: `tokens.basic`
     */
    private String token;

    /**
     * Specify `true` to convert `W` global user IDs to workspace-specific `U` IDs. Defaults to `false`.
     */
    private boolean toOld;

    /**
     * A comma-separated list of user ids, up to 400 per request
     */
    private List<String> users;

    MigrationExchangeRequest(String token, boolean toOld, List<String> users) {
        this.token = token;
        this.toOld = toOld;
        this.users = users;
    }

    public static MigrationExchangeRequestBuilder builder() {
        return new MigrationExchangeRequestBuilder();
    }

    public String getToken() {
        return this.token;
    }

    public boolean isToOld() {
        return this.toOld;
    }

    public List<String> getUsers() {
        return this.users;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public void setToOld(boolean toOld) {
        this.toOld = toOld;
    }

    public void setUsers(List<String> users) {
        this.users = users;
    }

    public static class MigrationExchangeRequestBuilder {
        private String token;
        private boolean toOld;
        private List<String> users;

        MigrationExchangeRequestBuilder() {
        }

        public MigrationExchangeRequest.MigrationExchangeRequestBuilder token(String token) {
            this.token = token;
            return this;
        }

        public MigrationExchangeRequest.MigrationExchangeRequestBuilder toOld(boolean toOld) {
            this.toOld = toOld;
            return this;
        }

        public MigrationExchangeRequest.MigrationExchangeRequestBuilder users(List<String> users) {
            this.users = users;
            return this;
        }

        public MigrationExchangeRequest build() {
            return new MigrationExchangeRequest(token, toOld, users);
        }

        public String toString() {
            return "MigrationExchangeRequest.MigrationExchangeRequestBuilder(token=" + this.token + ", toOld=" + this.toOld + ", users=" + this.users + ")";
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/migration/MigrationExchangeRequest.kt
package work.socialhub.kslack.api.methods.request.migration

import work.socialhub.kslack.api.methods.FormRequest
import work.socialhub.kslack.api.methods.SlackApiRequest
import kotlin.js.JsExport

/**
 * For Enterprise Grid workspaces, map local user IDs to global user IDs
 */
@JsExport
class MigrationExchangeRequest(
    /** Authentication token. Requires scope: `tokens.basic` */
    override var token: String?,
    /** Specify `true` to convert `W` global user IDs to workspace-specific `U` IDs. Defaults to `false`. */
    var isToOld: Boolean,
    /** A comma-separated list of user ids, up to 400 per request */
    var users: Array<String>?
) : SlackApiRequest, FormRequest {

    override fun toMap(): Map<String, Any> {
        return mutableMapOf<String, Any>().also {
            it.addParam("to_old", isToOld)
            if (users != null) {
                it.addParam("users", users!!.joinToString(","))
            }
        }
    }
}